### PR TITLE
FM2-283:  Add Integration Tests for AllergyIntolerance

### PR DESCRIPTION
**Pull request by varung-31**
## Description of what I changed

Add Integration Tests for AllergyIntolerance
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-283

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/AllergyIntoleranceFhirResourceProviderIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+		assertThat(entries, everyItem(
+		    hasResource(hasProperty("patient", hasProperty("identifier", hasProperty("value", equalTo("M4001-1")))))));
+		assertThat(entries,
+		    containsInRelativeOrder(hasResource(hasProperty("criticality", hasProperty("display", equalTo("Low Risk")))), // mild
+		        hasResource(hasProperty("criticality", hasProperty("display", equalTo("Unable to Assess Risk")))), // moderate
+		        hasResource(hasProperty("criticality", hasProperty("display", equalTo("High Risk")))), // severe
+		        hasResource(hasProperty("criticality", equalTo(null))))); // null
+		assertThat(entries, everyItem(hasResource(validResource())));
+		
+	}
```
Comment (corneliouzbett) [inline integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/AllergyIntoleranceFhirResourceProviderIntegrationTest.java:463]:
```suggestion
		assertThat(entries, everyItem(hasResource(validResource())));
	}
```

File: integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/AllergyIntoleranceFhirResourceProviderIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+	
+}
```
Comment (corneliouzbett) [inline integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/AllergyIntoleranceFhirResourceProviderIntegrationTest.java:511]:
```suggestion
		        hasResource(hasProperty("criticality", equalTo(null))))); // null
		assertThat(entries, everyItem(hasResource(validResource())));
	}
}
```

**Comment by ibacher** [comment]
@varung-31: Could you take a look at the conflicts here? I suspect the changes are very similar.

**Comment by varung-31** [comment]
@ibacher Why did you close this ?

**Comment by ibacher** [comment]
@varung-31 That was a mistake. Must've clicked the wrong button.

**Comment by varung-31** [comment]
@ibacher Pushed the changes

**Review by ibacher — APPROVED** [review]

**Pull request merged**